StyleCop.Analyzers/StyleCop.Analyzers.Test/SpacingRules/SA1027UnitTests.cs

```diff
@@ -0,0 +1,99 @@
+﻿namespace StyleCop.Analyzers.Test.SpacingRules
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis.CodeFixes;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using StyleCop.Analyzers.SpacingRules;
+    using TestHelper;
+    using Xunit;
+
+    /// <summary>
+    /// Unit tests for <see cref="SA1027TabsMustNotBeUsed"/>
+    /// </summary>
+
+    public class SA1027UnitTests : CodeFixVerifier
+    {
+        /// <summary>
+        /// Verifies that the analyzer will properly handle an empty source.
+        /// </summary>
+        [Fact]
+        public async Task TestEmptySource()
+        {
+            var testCode = string.Empty;
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None);
+        }
+
+        /// <summary>
+        /// Verifies that tabs used inside string and char literals are not producing diagnostics.
+        /// </summary>
+        [Fact]
+        public async Task TestValidTabs()
+        {
+            var testCode =
+                "public class Foo\r\n" +
+                "{\r\n" +
+                "    public const string ValidTestString = \"\tText\";\r\n" +
+                "    public const char ValidTestChar = '\t';\r\n" +
+                "}\r\n";
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None);
+        }
+
+        [Fact]
+        public async Task TestInvalidTabs()
+        {
+            var testCode =
+                "using\tSystem.Diagnostics;\r\n" +
+                "\r\n" +
+                "public class Foo\r\n" +
+                "{\r\n" +
+                "\tpublic void Bar()\r\n" +
+                "\t{\r\n" +
+                "\t  \t// Comment\r\n" +
+                "\t \tDebug.Indent();\r\n" +
+                "   \t}\r\n" +
+                "}\r\n";
+
+            // There is only a single space between the using and the System.Diagnostics, 
```

Comment:
I don't agree. Inserting three spaces there will indeed keep the original layout, but it will immediately trigger SA1025, something that is not desirable in my opinion.